scipy/special/tests/test_legendre.py

```diff
@@ -678,7 +687,11 @@ def test_ode(self, shape):
 class TestLegendreFunctions:
     def test_clpmn(self):
         z = 0.5+0.3j
-        clp = special.clpmn(2, 2, z, 3)
+
+        with suppress_warnings() as sup:
+            sup.filter(category=DeprecationWarning)
+            clp = special.clpmn(2, 2, z, 3)
+
         assert_array_almost_equal(clp,
                    (np.array([[1.0000, z, 0.5*(3*z*z-1)],
                            [0.0000, np.sqrt(z*z-1), 3*z*np.sqrt(z*z-1)],
```

Comment:
I don't know if or to what degree these tests were duplicated when introducing the new implementation, but if for some reason these tests are unique w.r.t. what aspect of `clpmn` resp. `assoc_legendre_p_all` they test, then we should simply update those tests to use the new function names, rather than sprinkling ignored warnings in lots of places.

Reply:
Yea I think that makes sense, but it might be a bit out-of-scope to do this within this PR.

Reply:
Good thinking. What's going on though is that e.g. `assoc_legendre_p_all` isn't just a drop in replacement `clpmn`. The new functions provide a new, more featureful, and more consistent interface for Legendre functions.

`clpmn` calls `assoc_legendre_p_all` under the hood, but the process to convert to the right output type isn't exactly trivial:

https://github.com/scipy/scipy/blob/ba8553c12cfa4d3ee3df1f862230f360638dd64c/scipy/special/_basic.py#L1856-L1867

so we left the old tests in to make sure the conversion is correct. I think it's fine to keep them around until 1.17.0, but it would be worth having comments explaining why they are there.

Reply:
OK, sounds good! The comments don't have to come as part of this PR.